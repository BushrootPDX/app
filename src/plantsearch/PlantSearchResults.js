import React, { Component } from 'react';
import { connect } from 'react-redux';
import { selectPlant } from './actions';

export class PlantSearchResults extends Component {

    render() {

        const { plants, selectPlant } = this.props;

        return (
            <ul>
                {plants && plants.map((plant, index) => 
                    (<li key={index} 
                        id={plant._id} 
                        onSelect={() => selectPlant(plant._id)}>
                        {plant.name}
                    </li> )
                )}
            </ul>
        );
    }
}

export default connect(
    state => ({ 
        plants: state.plants, 
    }),
    dispatch => ({ 
        selectPlant: dispatch(selectPlant()) 
    })
)(PlantSearchResults);