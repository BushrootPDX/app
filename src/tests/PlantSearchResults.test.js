import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PlantSearchResults from '../plantsearch/PlantSearchResults';

const results = [{
    name: 'bill',
    id: '123'
}, {
    name: 'steve',
    id: '456'
}, {
    name: 'tomato',
    id: 'red'
}];
const selectPlant = function(id) {
    console.log(id);
};


describe('PlantSearchResults component', () => {

    it('renders the plant search results when given as an array', () => {
        const wrapper = shallow(PlantSearchResults(results, selectPlant));
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});