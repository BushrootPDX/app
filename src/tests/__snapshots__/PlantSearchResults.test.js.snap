// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantSearchResults component renders the plant search results when given as an array 1`] = `<ul />`;
